### DISCUSSION TITLE
graphql/hive/docs/get-started/apollo-federation

### DISCUSSION THREAD

**Discussion by giscus** [graphql-hive/console#5612] [Docs Discussions]
# graphql/hive/docs/get-started/apollo-federation

Hive Documentation

https://the-guild.dev/graphql/hive/docs/get-started/apollo-federation

**Comment by jenskm** [comment]
The `Person` type within "subgraphs/reviews.graphql" is defined as follows:

```graphql
extend type User @key(fields: "id") {
  id: ID! @external
  username: String @external
  reviews: [Review]
}
```

Upon attempting to publish this subgraph to Hive, these errors occur:

```plaintext
Composition Errors

[reviews] Type User is an extension type, but there is no type definition for User in any subgraph.
Field User.username is marked @external on all the subgraphs in which it is listed (subgraph reviews).
```

Additionally, the screenshot associated with the statement, "If you’ll get to your target’s page on Hive dashboard, you should see that it was updated with the new schema you just published," displays types such as `Price`, `Money`, and `Department`. However, these types are not present in the products schema.

This leads to the question: Are one, or both, of the provided example schema files incomplete?

**Reply by n1ru4l** [reply]
Hey @jenskm, thank you for reporting this. Indeed, the user subgraph was missing. 😓
I just worked on correcting this mistake in the documentation over [here](https://github.com/kamilkisiela/graphql-hive/pull/5630). We will deploy the updated documentation soon.

**Comment by LongLiveCHIEF** [comment]
I am a bit confused around why the `extend` keyword is needed in the second subgraph, even though the type is an entity?
